components/server/src/user/user-controller.ts

```diff
@@ -601,15 +601,22 @@ export class UserController {
     }
 
     protected getSafeReturnToParam(req: express.Request) {
-        const returnToURL: string | undefined = req.query.redirect || req.query.returnTo;
-        if (returnToURL) {
-            const hostUrl = this.env.hostUrl.url as URL;
-            if (returnToURL.toLowerCase().startsWith(`${hostUrl.protocol}//${hostUrl.host}`.toLowerCase())) {
```

Comment:
@leodido if you are looking for a quick fix here, a change that ensures a trailing slash of the host name should work as well. 
```
returnToURL.toLowerCase().startsWith(`${hostUrl.protocol}//${hostUrl.host}/`.toLowerCase())
```